topology/pynest/tests/test_spatial_kernels.py

```diff
@@ -0,0 +1,780 @@
+# -*- coding: utf-8 -*-
+#
+# test_spatial_kernels.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Tests distribution of connections created with spatial kernels.
+
+Original code by Daniel Hjertholm with Birgit Kriener.
+Converted to automated test by Hans Ekkehard Plesser.
+
+For theory, see
+D. Hjertholm, Statistical tests for connection algorithms for
+structured neural networks, MSc thesis, Norwegian University of
+Life Science, 2013. http://hdl.handle.net/11250/189117.
+"""
+
+import math
+import numpy as np
+import numpy.random as rnd
+import scipy.integrate
+import scipy.stats
+import scipy.special
+import unittest
+
+import nest
+import nest.topology as topo
+
+
+try:
+    # for debugging
+    from mpl_toolkits.mplot3d import Axes3D
+    import matplotlib.pyplot as plt
+    # make sure we can open a window; DISPLAY may not be set
```

Comment:
What does DISPLAY refer to?

Reply:
It's an environment variable for X11 programs. You can check yours in terminal.
`echo $DISPLAY

In this case matplotlib needs to have DISPLAY environment for opening a window.